connectors/sources/sharepoint_online.py

```diff
@@ -282,14 +283,11 @@ async def wrapped(*args, **kwargs):
                     async for item in func(*args, **kwargs):
                         yield item
                     break
-                except (
-                    ThrottledError,
-                    PermissionsMissing,
-                    InternalServerError,
-                ) as e:
+                except (NotFound, ClientResponseError):
+                    raise
+                except Exception:
```

Comment:
nit: It would be also useful to handle `ClientOSError` and log the error that MS API closed the session.

Reply:
Do you still want to retry it?

Reply:
Do you still want to retry it?

I think the fact that MS API closed the session is a sign, that connector is overthrottling and can potentially be banned from Graph API access actually. I'd probably throw with a nice error message